File: Python3/1331. Rank Transform of an Array.py
class Solution:
    def arrayRankTransform(self, arr: List[int]) -> List[int]:
        nums = list(set(arr))
        nums.sort()

        map = {num: rank + 1 for rank, num in enumerate((nums))}
        for i, a in enumerate(arr):
            arr[i] = map[a]

        return arr

File: Java/1331. Rank Transform of an Array.java
class Solution {
    public int[] arrayRankTransform(int[] arr) {
        Map<Integer, Integer> map = new HashMap<>();
        TreeSet<Integer> nums = new TreeSet<>();
        for (int n: arr) {
            nums.add(n);
        }

        int rank = 1;
        for (int n: nums) {
            map.put(n, rank++);
        }

        for (int i = 0; i < arr.length; i++) {
            arr[i] = map.get(arr[i]);
        }

        return arr;
    }
}